### PR TITLE
[LM-11319] Added view participants screen

### PR DESCRIPTION
**Pull request by AnujLM**
## Description

**Features**
1. Added view participants screen

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: lib/packages/core/lib/src/blocs/participants/handler/get_participants_handler.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
+part of "../participants_bloc.dart";
+
+void _getParticipantsEventHandler(LMChatGetParticipantsEvent event,
+    Emitter<LMChatParticipantsState> emit) async {
+  emit(const LMChatParticipantsLoadingState());
+  try {
+    final GetParticipantsRequest getParticipantsRequest =
```
Comment (divyanshgandhilm) [inline lib/packages/core/lib/src/blocs/participants/handler/get_participants_handler.dart:7]: Try to add comments here as well, if possible.

File: lib/packages/core/lib/src/blocs/participants/participants_state.dart [hunk 2]
```diff
@@ -1,32 +1,48 @@
...
 
-class LMChatParticipantsLoaded extends LMChatParticipantsState {
-  final GetParticipantsResponse getParticipantsResponse;
+  /// [page] is the current page for pagination.
+  final int page;
 
-  const LMChatParticipantsLoaded({required this.getParticipantsResponse});
+  /// [LMChatParticipantsLoadedState] constructor to create an instance of [LMChatParticipantsLoadedState].
+  const LMChatParticipantsLoadedState(
+      {required this.participants, required this.page});
```
Comment (divyanshgandhilm) [inline lib/packages/core/lib/src/blocs/participants/participants_state.dart:34]: Add `,` for formatting params

File: lib/packages/ui/lib/src/utils/typedefs.dart [hunk 3]
```diff
@@ -1,38 +1,111 @@
 import 'package:flutter/material.dart';
 import 'package:likeminds_chat_flutter_ui/likeminds_chat_flutter_ui.dart';
```
Comment (divyanshgandhilm) [inline lib/packages/ui/lib/src/utils/typedefs.dart:2]: We have to import the files inside `src` directory. Can't directly import the library inside itself. Remove from everywhere.

File: lib/packages/ui/lib/src/widgets/common/tile/user_tile.dart [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+import 'package:likeminds_chat_flutter_ui/likeminds_chat_flutter_ui.dart';
+
+class LMChatUserTile extends StatelessWidget {
```
Comment (divyanshgandhilm) [inline lib/packages/ui/lib/src/widgets/common/tile/user_tile.dart:4]: Should extend the `LMChatTile` class and provide customisations through its style class.

File: lib/packages/core/lib/src/views/participants/participants.dart [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+  final Widget Function(BuildContext)? noItemsFoundIndicatorBuilder;
+
+  /// [noMoreItemsIndicatorBuilder] is a builder to show the indicator when no more participants are found.
+  final Widget Function(BuildContext)? noMoreItemsIndicatorBuilder;
+
+  /// [appBarBuilder] is a builder to build the app bar for the page.
+  final LMChatAppBarBuilder? appBarBuilder;
+
+  /// [userTileBuilder] is a builder to build the user tile for the participants.
+  final Widget Function(BuildContext, LMChatUserTile)? userTileBuilder;
```
Comment (divyanshgandhilm) [inline lib/packages/core/lib/src/views/participants/participants.dart:37]: We should use a typedef here as well. But of type `LMChatTile`.

File: lib/packages/ui/lib/src/widgets/profile/user_tile.dart [hunk 6]
```diff
@@ -3,79 +3,58 @@ import 'package:likeminds_chat_flutter_ui/src/models/models.dart';
 import 'package:likeminds_chat_flutter_ui/src/theme/theme.dart';
 import 'package:likeminds_chat_flutter_ui/src/widgets/widgets.dart';
 
+/// [LMChatUserTile] is a [LMChatTile] that represents a user in a chat room.
 class LMChatUserTile extends LMChatTile {
```
Comment (divyanshgandhilm) [inline lib/packages/ui/lib/src/widgets/profile/user_tile.dart:7]: This one is correct, are we not using this?

**Review by divyanshgandhilm — CHANGES_REQUESTED** [review]

File: lib/packages/core/lib/src/blocs/participants/handler/get_participants_handler.dart (hunk 1 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/blocs/participants/handler/get_participants_handler.dart:7]: done

File: lib/packages/core/lib/src/blocs/participants/participants_state.dart (hunk 2 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/blocs/participants/participants_state.dart:34]: done

File: lib/packages/ui/lib/src/widgets/common/tile/user_tile.dart (hunk 4 above)
Comment (AnujLM) [reply lib/packages/ui/lib/src/widgets/common/tile/user_tile.dart:4]: It was redundant as we are using LMChatTile from `/src/widgets/profile/user_tile.dart`

File: lib/packages/core/lib/src/views/participants/participants.dart (hunk 5 above)
Comment (AnujLM) [reply lib/packages/core/lib/src/views/participants/participants.dart:37]: done

File: lib/packages/ui/lib/src/utils/typedefs.dart (hunk 3 above)
Comment (AnujLM) [reply lib/packages/ui/lib/src/utils/typedefs.dart]: done

**Review by divyanshgandhilm — APPROVED** [review]

**Pull request merged**